execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -241,7 +241,7 @@ where
     E: From<S::Error> + From<bytesrepr::Error>,
 {
     let mut missing_descendants = Vec::new();
-    let mut trie_keys_to_visit_queue = vec![trie_key];
+    let mut trie_keys_to_visit_queue = trie_key;
```

Comment:
Because we can have tries with common children, this should be be a `HashSet`, not a `Vec`.

We could also consider renaming `trie_key` and `trie_keys_to_visit_queue` to `trie_keys_to_visit`, since it's a bit confusing they way they are currently named.